src/ripple/app/paths/impl/BookStep.cpp

```diff
@@ -533,10 +533,10 @@ BookStep<TIn, TOut, TDerived>::forEachOffer (
                 // Issuer requires authorization.  See if offer owner has that.
                 auto const& ownerID = offer.owner();
                 auto const authFlag =
-                    ownerID > issuerID ? lsfHighAuth : lsfLowAuth;
+                    ownerID > issuerID ? lsfLowAuth : lsfHighAuth;
 
                 auto const line = afView.read (keylet::line (
-                    ownerID, issuerID, offer.issueOut().currency));
+                    ownerID, issuerID, offer.issueIn().currency));
```

Comment:
I see.  I agree that line 530 check protects it.  But if that weren't around, I think the offer would be deleted because line 541 deletes on a missing trustline or an authorized one.